src/backend/distributed/commands/create_citus_local_table.c

```diff
@@ -202,6 +202,8 @@ ErrorIfUnsupportedCreateCitusLocalTable(Relation relation)
 							   "not exist")));
 	}
 
+	ErrorIfTableIsACatalogTable(relation);
```

Comment:
I think this is a common thing for all types of tables in citus right? Like there are certain things that we cannot do with any type of citus table. I think it could be better to include such things in a common method which is called before each type of citus table creation, that way we have only one place to control that and it is easier to maintain. What do you think?

Reply:
I was thinking of exactly the same, maybe in a seperate pr I can do that ?